src/nvoptix_22.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2020 NVIDIA Corporation.  All rights reserved.
+ *
+ * NVIDIA Corporation and its licensors retain all intellectual property and proprietary
+ * rights in and to this software, related documentation and any modifications thereto.
+ * Any use, reproduction, disclosure or distribution of this software and related
+ * documentation without an express license agreement from NVIDIA Corporation is strictly
+ * prohibited.
+ *
+ * TO THE MAXIMUM EXTENT PERMITTED BY APPLICABLE LAW, THIS SOFTWARE IS PROVIDED *AS IS*
+ * AND NVIDIA AND ITS SUPPLIERS DISCLAIM ALL WARRANTIES, EITHER EXPRESS OR IMPLIED,
+ * INCLUDING, BUT NOT LIMITED TO, IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE.  IN NO EVENT SHALL NVIDIA OR ITS SUPPLIERS BE LIABLE FOR ANY
+ * SPECIAL, INCIDENTAL, INDIRECT, OR CONSEQUENTIAL DAMAGES WHATSOEVER (INCLUDING, WITHOUT
+ * LIMITATION, DAMAGES FOR LOSS OF BUSINESS PROFITS, BUSINESS INTERRUPTION, LOSS OF
+ * BUSINESS INFORMATION, OR ANY OTHER PECUNIARY LOSS) ARISING OUT OF THE USE OF OR
+ * INABILITY TO USE THIS SOFTWARE, EVEN IF NVIDIA HAS BEEN ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGES
+ */
+
+#pragma once
+
+#include "nvoptix.h"
+#include <stddef.h>
+
+// defensive duplicate of OptixDeviceContextOptions because I have to modify it
+
+typedef void (*OptixLogCallback_22)(unsigned int level, const char *tag, const char *message, void *cbdata);
+
+typedef struct OptixDeviceContextOptions_22
+{
+    OptixLogCallback_22 logCallbackFunction;
+    void *logCallbackData;
+    int logCallbackLevel;
+    int validationMode;
```

Comment:
Are you sure this is correct? Public headers available at https://developer.download.nvidia.com/redist/optix/v7.0/OptiX-7.0.0-include.zip claim that this struct only has first three members.

Reply:
Does this look reasonable?
https://github.com/SveSop/wine-nvoptix/commit/e6760455e4c97282e09e4c5d1450e8997abe1893

Reply:
Yeah, should be good now.

Reply:
I wonder if this log business could be fixed somehow... It seems as Iray/DAZ is using this for tracing error messages since it will break down miserably if just pass it through without modifications.
The simple demo stuff happily lets this pass with a setting of "3" for `LogCallbackLevel` but it probably does not really use this for anything anyway.

If it is a pointer to some message array someplace in memory, it might not fly or smt.. Well, thats for another time i guess.

Reply:
I can take a look at this, but it probably won't be pretty.